staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -1761,20 +1761,22 @@ func sortScalars(s []interface{}) []interface{} {
 }
 
 func deduplicateScalars(s []interface{}) []interface{} {
-	// Clever algorithm to deduplicate.
-	length := len(s) - 1
-	for i := 0; i < length; i++ {
-		for j := i + 1; j <= length; j++ {
-			if s[i] == s[j] {
-				s[j] = s[length]
-				s = s[0:length]
-				length--
-				j--
-			}
-		}
-	}
-
-	return s
+	m := make(map[interface{}]bool)
```

Comment:
for small lists, the memory cost of this allocation is higher. we need benchmarks around lists of various sizes to make sure this is actually a net win